--- a/qtbase/mkspecs/win32-g++/qmake.conf
+++ b/qtbase/mkspecs/win32-g++/qmake.conf
@@ -27,4 +27,10 @@ QMAKE_CFLAGS_LTCG       = -flto
 QMAKE_CXXFLAGS_LTCG     = $$QMAKE_CFLAGS_LTCG
 QMAKE_LFLAGS_LTCG       = $$QMAKE_CFLAGS_LTCG
 
+# [QT-STATIC-PATCH]
+QMAKE_LFLAGS           += -static -static-libgcc
+QMAKE_CFLAGS_RELEASE   -= -O2
+QMAKE_CFLAGS_RELEASE   += -Os -momit-leaf-frame-pointer
+DEFINES                += QT_STATIC_BUILD
+
 load(qt_config)

